client/blocks/upsell-nudge/index.jsx

```diff
@@ -192,3 +224,16 @@ export default connect( ( state, ownProps ) => {
 		siteIsWPForTeams: isSiteWPForTeams( state, getSelectedSiteId( state ) ),
 	};
 } )( UpsellNudge );
+
+export default function Wrapper( props ) {
+	if ( props.isOneClickCheckoutEnabled ) {
+		return (
+			<AsyncLoad
+				require="../../my-sites/checkout/purchase-modal/is-eligible-for-one-click-checkout-wrapper"
+				component={ ConnectedUpsellNudge }
+				componentProps={ props }
+			/>
+		);
+	}
+	return <ConnectedUpsellNudge { ...props } />;
+}
```

Comment:
The `useIsEligibleForOneClickCheckout` uses checkout and payment-related code, which added to the bundle size of entry sections significantly. This conditional loading was required to ensure that the bundle size issue does not happen.